app/main/forms/suppliers.py

```diff
@@ -50,7 +50,9 @@ class DunsNumberForm(Form):
 class CompaniesHouseNumberForm(Form):
     companies_house_number = StripWhitespaceStringField('Companies house number', validators=[
         Optional(),
-        Length(min=8, max=8, message="Companies House numbers must have 8 characters.")
+        Regexp(r'^([0-9]{2}|[A-Za-z]{2})[0-9]{6}$',
+               message="Companies House numbers must have 8 digits or 2 letters followed by 6 digits."
```

Comment:
Could we encode the rule as a regexp directly: `[0-9]{8} | [A-Za-z]{2}[0-9]{6}`. It's a bit longer, but it does exactly what the message says, which I think makes it easier to parse.

The rest looks good!

Reply:
I can see your point but this exactly matches the regex already used in the API, so might be nice to keep them the same? What do you reckon?  Could change in both places, but... pull requests.

Reply:
Oh, yeah, then keep it as is. I guess the message might change as well and it doesn't describe the format that much better either.